compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeBuilder.java

```diff
@@ -4950,26 +4951,98 @@ BLangValueType addValueType(Location pos, TypeKind typeKind) {
-        return generateAndAddBLangStatements(statementNodes, statements);
+        return generateAndAddBLangStatements(statementNodes, statements, endNode);
     }
 
     private List<BLangStatement> generateAndAddBLangStatements(NodeList<StatementNode> statementNodes,
-                                                               List<BLangStatement> statements) {
+                                                               List<BLangStatement> statements, Node endNode) {
+        int c = -1;
         for (StatementNode statement : statementNodes) {
+            c++;
             // TODO: Remove this check once statements are non null guaranteed
             if (statement != null) {
                 if (statement.kind() == SyntaxKind.FORK_STATEMENT) {
                     generateForkStatements(statements, (ForkStatementNode) statement);
                     continue;
                 }
+                if (statement.kind() == SyntaxKind.IF_ELSE_STATEMENT &&
+                        ((IfElseStatementNode) statement).elseBody().isEmpty()) {
+                    statements.add((BLangStatement) statement.apply(this));
+                    if (c == statementNodes.size() - 1) {
+                        statements.add((BLangBlockStmt) TreeBuilder.createBlockNode());
+                        return statements;
+                    }
+                    if (c < statementNodes.size() - 1) {
+                        BLangBlockStmt bLBlockStmt = (BLangBlockStmt) TreeBuilder.createBlockNode();
+                        this.isInLocalContext = true;
+                        StatementNode remainingStatement = statementNodes.get(c + 1);
+                        if (statementNodes.get(c + 1) != null) {
+                            createBlockStatement(remainingStatement, c + 1, statementNodes, bLBlockStmt.stmts,
+                                    endNode);
+                            this.isInLocalContext = false;
+                            if (bLBlockStmt.stmts.isEmpty()) {
+                                return statements;
+                            }
+                            if (c + 1 <= statementNodes.size() - 1) {
+                                bLBlockStmt.pos = getPosition(statementNodes.get(c + 1),
+                                        endNode);
+                            }
+                            statements.add(bLBlockStmt);
+                        }
+                    }
+                    return statements;
+                }
                 statements.add((BLangStatement) statement.apply(this));
             }
         }
         return statements;
     }
 
+    private void createBlockStatement(StatementNode currentStatement, int position,
+                                      NodeList<StatementNode> statementNodes,
+                                      List<BLangStatement> statements, Node endNode) {
+        for (int j = position; j < statementNodes.size(); j++) {
```

Comment:
Can you check whether merging the logics in the two methods is feasible since we are doing the same thing.

Reply:
suggestions in generateAndAddBLangStatements() are applicable here as well